### ISSUE TITLE
[Bug] Please fill in your issue title here.

### ISSUE THREAD

**Issue by OEvortex** [OEvortex/unsloth#2]
i have 4H100 gpus but multiple gpu training not working plz fix that 
pythonn3.11) ubuntu@192-222-55-138:~$ python gpu.py
🦥 Unsloth: Will patch your computer to enable 2x faster free finetuning.
🦥 Unsloth Zoo will now patch everything to make training faster!
==((====))==  Unsloth 2025.7.11: Fast Qwen3 patching. Transformers: 4.54.1.
   \\   /|    NVIDIA H100 80GB HBM3. Num GPUs = 4. Max memory: 79.189 GB. Platform: Linux.
O^O/ \_/ \    Torch: 2.7.1+cu126. CUDA: 9.0. CUDA Toolkit: 12.6. Triton: 3.3.1
\        /    Bfloat16 = TRUE. FA [Xformers = None. FA2 = False]
 "-____-"     Free license: http://github.com/unslothai/unsloth
Unsloth: Fast downloading is enabled - ignore downloading bars which are red colored!
model.safetensors.index.json: 32.8kB [00:00, 149MB/s]
model-00001-of-00002.safetensors: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 4.97G/4.97G [00:03<00:00, 1.28GB/s]
model-00002-of-00002.safetensors: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 3.08G/3.08G [00:02<00:00, 1.19GB/s]
Loading checkpoint shards: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 2/2 [00:01<00:00,  1.46it/s]
generation_config.json: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 237/237 [00:00<00:00, 2.90MB/s]
tokenizer_config.json: 10.5kB [00:00, 53.9MB/s]
vocab.json: 2.78MB [00:00, 83.0MB/s]
merges.txt: 1.67MB [00:00, 132MB/s]
added_tokens.json: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 707/707 [00:00<00:00, 9.50MB/s]
special_tokens_map.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 614/614 [00:00<00:00, 8.12MB/s]
tokenizer.json: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11.4M/11.4M [00:00<00:00, 27.9MB/s]
chat_template.jinja: 4.76kB [00:00, 27.5MB/s]
Unsloth 2025.7.11 patched 36 layers with 36 QKV layers, 36 O layers and 36 MLP layers.
README.md: 9.15kB [00:00, 43.2MB/s]
1_filtered.json: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 78.2M/78.2M [00:00<00:00, 78.3MB/s]
Generating train split: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 11720/11720 [00:01<00:00, 10653.81 examples/s]
<|im_start|>user
Who was responsible for the assassination of Archduke Franz Ferdinand in 1914?<|im_end|>
<|im_start|>assistant
<think>
This is a historical question about the assassination of Archduke Franz Ferdinand, which was a pivotal event that helped trigger World War I. I need to provide accurate information about who was responsible. Gavrilo Princip was the actual assassin who shot the Archduke and his wife. He was a Bosnian Serb nationalist and member of the revolutionary group Young Bosnia. The assassination was supported by the Black Hand, a Serbian nationalist organization. I should provide context about the political situation and motivations.
</think>

While Princip pulled the trigger, he wasn't acting alone. The assassination was supported by the Black Hand (officially \"Unification or Death\"), a Serbian nationalist secret society led by Colonel Dragutin Dimitrijević, codenamed \"Apis.\" The Black Hand provided the weapons and training for the assassination.

There were actually six conspirators positioned along the Archduke's route through Sarajevo that day. One conspirator, Nedeljko Čabrinović, had earlier thrown a bomb at the Archduke's car, but it bounced off and exploded under the following vehicle. It was only later, after a wrong turn by the Archduke's driver, that Princip found himself with an unexpected opportunity to carry out the assassination at point-blank range.

The killing set off a chain of diplomatic events that, within weeks, led to the outbreak of World War I as the complex system of European alliances and tensions finally collapsed into open conflict.<|im_end|>

11720
Unsloth: Tokenizing ["text"] (num_proc=2): 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████| 11720/11720 [00:06<00:00, 1755.39 examples/s]
GPU = NVIDIA H100 80GB HBM3. Max memory = 79.189 GB.
1.938 GB of memory reserved.
==((====))==  Unsloth - 2x faster free finetuning | Num GPUs used = 4
   \\   /|    Num examples = 11,720 | Num Epochs = 1 | Total steps = 1,465
O^O/ \_/ \    Batch size per device = 2 | Gradient accumulation steps = 4
\        /    Data Parallel GPUs = 1 | Total batch size (2 x 4 x 1) = 8
 "-____-"     Trainable parameters = 66,060,288 of 4,088,528,384 (1.62% trained)
  0%|                                                                                                                                                                           | 0/1465 [00:00<?, ?it/s]Unsloth: Will smartly offload gradients to save VRAM!
{'loss': 6.9831, 'grad_norm': 21.8949031829834, 'learning_rate': 0.0, 'epoch': 0.0}                                                                                                                      
{'loss': 5.7734, 'grad_norm': 8.700607299804688, 'learning_rate': 4e-05, 'epoch': 0.0}                                                                                                                   

(pythonn3.11) ubuntu@192-222-55-138:~$ nvidia-smi
Fri Aug  1 11:14:23 2025       
+-----------------------------------------------------------------------------------------+
| NVIDIA-SMI 570.148.08             Driver Version: 570.148.08     CUDA Version: 12.8     |
|-----------------------------------------+------------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id          Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |           Memory-Usage | GPU-Util  Compute M. |
|                                         |                        |               MIG M. |
|=========================================+========================+======================|
|   0  NVIDIA H100 80GB HBM3          On  |   00000000:06:00.0 Off |                    0 |
| N/A   28C    P0             69W /  700W |       0MiB /  81559MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   1  NVIDIA H100 80GB HBM3          On  |   00000000:07:00.0 Off |                    0 |
| N/A   29C    P0             74W /  700W |       0MiB /  81559MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   2  NVIDIA H100 80GB HBM3          On  |   00000000:08:00.0 Off |                    0 |
| N/A   31C    P0             71W /  700W |       0MiB /  81559MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
|   3  NVIDIA H100 80GB HBM3          On  |   00000000:09:00.0 Off |                    0 |
| N/A   30C    P0             70W /  700W |       0MiB /  81559MiB |      0%      Default |
|                                         |                        |             Disabled |
+-----------------------------------------+------------------------+----------------------+
                                                                                         
+-----------------------------------------------------------------------------------------+
| Processes:                                                                              |
|  GPU   GI   CI              PID   Type   Process name                        GPU Memory |
|        ID   ID                                                               Usage      |
|=========================================================================================|
|  No running processes found                                                             |
+-----------------------------------------------------------------------------------------+
(pythonn3.11) ubuntu@192-222-55-138:~$
